File: flowers-control-app/src/models/deviceDetail.ts
import { Status } from "./deviceItem";


export type DeviceDetailType = {
    sensors: Sensor[];
	id: string,
    name: string,
    status: Status
}

export type Sensor = {

    id: string,

    name: string,

    sensorType: SensorType,

    status: SensorStatus,

    data: SensorDataType,

	warnings: string[]
};

export type SensorDataType = {
    humidity?:number
}

export enum SensorType {
    FlowerHumiditySensor
};

export enum SensorStatus {
    Offline,
    Online
}

File: devices-api/Models/DeviceDetail.cs
public class DeviceDetail
{
	public string? Name { get; set; }
	public string? Id { get; set; }
	public DeviceStatus Status { get; set; }
	public IEnumerable<Sensor>? Sensors { get; set; }
	public List<string>? Admins {get;set;}
	public List<string>? Guests {get;set;}
}